hy/compiler.py

```diff
@@ -379,8 +386,9 @@ def _compile_collect(self, exprs, with_kwargs=False, dict_display=False,
 
             if not PY35 and oldpy_unpack and is_unpack("iterable", expr):
                 if oldpy_starargs:
-                    raise HyTypeError(expr, "Pythons < 3.5 allow only one "
-                                            "`unpack-iterable` per call")
+                    raise HyTypeError("Pythons < 3.5 allow only one "
+                                      "`unpack-iterable` per call",
+                                      (self.filename, expr, self.source))
```

Comment:
Having to say ` (self.filename, expr, self.source)` every time is a drag. Try a helper function or method, like

    def err(self, expr, msg):
        return HyTypeError(msg, (self.filename, expr, self.source))

Reply:
Yeah, I have have something like that in a correction I'm about to push.